pkg/services/govmomi/service.go

```diff
@@ -224,6 +224,16 @@ func (vms *VMService) reconcileMetadata(ctx *virtualMachineContext) (bool, error
 		return false, err
 	}
 
+	// checking that the vsphereVM's network statuses, if any match the number of
+	// network devices for the VSphereVM
+	if statuses := len(ctx.State.Network); statuses > 0 {
+		if statuses != len(ctx.VSphereVM.Spec.Network.Devices) {
```

Comment:
Ideally, I'd like us to fail only when we remove a device. I don't want us to block additions, until we support things like `sr-iov`

Reply:
Adding a greater than check `>` should be sufficient in this case then.